x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/TopNOperator.java

```diff
@@ -394,6 +419,28 @@ public void addInput(Page page) {
         }
     }
 
+    /**
+     * Calculates the partition key of the i-th row of the given page.
+     *
+     * @param page page for which the partition key should be calculated
+     * @param i row index
+     * @return partition key of the i-th row of the given page
+     */
+    private String getPartitionKey(Page page, int i) {
+        if (partitions.isEmpty()) {
+            return "";
+        }
+        assert page.getPositionCount() > 0;
+        StringBuilder builder = new StringBuilder();
+        for (Partition partition : partitions) {
+            try (var block = page.getBlock(partition.channel).filter(i)) {
```

Comment:
I feel like `filter(i)` and `getBytesRef(i)` sort of duplicates this. I think you just need the latter. If you use `TopNEncoder.DEFAULT_UNSORTABLE` you can copy out of any block into a `BreakingBytesRefBuilder`.

Reply:
> I feel like filter(i) and getBytesRef(i) sort of duplicates this. I think you just need the latter.

Done.

> If you use TopNEncoder.DEFAULT_UNSORTABLE

What does it mean? Can't I just use `BreakingBytesRefBuilder` instead of `StringBuilder` here? (PR updated with this change).